lib/URI/_server.pm

```diff
@@ -5,19 +5,25 @@ use warnings;
 
 use parent 'URI::_generic';
 
-use URI::Escape qw(uri_unescape);
+use URI::Escape qw(uri_unescape uri_escape);
 
 our $VERSION = '5.29';
 
 sub _uric_escape {
     my($class, $str) = @_;
-    if ($str =~ m,^((?:$URI::scheme_re:)?)//([^/?\#]*)(.*)$,os) {
-	my($scheme, $host, $rest) = ($1, $2, $3);
-	my $ui = $host =~ s/(.*@)// ? $1 : "";
-	my $port = $host =~ s/(:\d+)\z// ? $1 : "";
-	if (_host_escape($host)) {
-	    $str = "$scheme//$ui$host$port$rest";
-	}
+    if ($str =~ m,^((?:$URI::scheme_re:)?)//(.*:.*@)?([^/?\#]*)(.*)$,os) {
+        my $scheme = $1;
+        my $ui = $2 || '';
```

Comment:
Could we have a more descriptive name for this variable?

Reply:
I realize this variable was not your creation. :)

Reply:
Sure thing, I have renamed to `$userinfo`.
I decided to rename in whole file as it might be a little confusing when we're using `$ui` or `$userinfo`.